### ISSUE TITLE
For CI - frequent corrupted cache due to Travis intermittent network infra problem

### ISSUE THREAD

**Issue by weitjong** [urho3d/Urho3D#2345]
See https://github.com/travis-ci/travis-ci/issues/9757.

If the problem cannot be resolved by Travis-CI team, we may have to find other workaround solution.

**Comment by weitjong** [comment]
Without the "Net Neutrality", I think we (read non-paying customers) need to teach our scripts how to do retry.

**Comment by ArnisLielturks** [comment]
What's the plan? I have noticed these problems a lot lately which kinda gives me idea that there's something wrong with my code. It works sorta like demotivation tool.

**Comment by weitjong** [comment]
I hope the issue will go away after Travis team fix the network on their side. Personally I will only spend as little time as possible to maintain the current setup. However, I will support other core developers who want to explore other possibility, even venturing outside of what being offered by Travis.org. It is an opportunity to have a new and better setup somewhere else. This issue could be a blessing in disguise.

**Comment by ArnisLielturks** [comment]
I've been playing around with CircleCI for some time now and I use it for all of my projects to automate builds. It does take a lot of time to rewrite everything what's already done for the TravisCi tool but small part of if I have already succesfully moved to CircleCI like MinGW and Linux builds, I will also try to get Android builds up and running in the nearest future but at a current pace it would take me few months to completely replace it.

**Comment by weitjong** [comment]
I come across this https://semaphoreci.com/. It looks promising. Although it only offers 2 free concurrent build as opposed to 5 offered free by Travis, but if what it claims is true (2x speed) then that could easily offset it. Plus it has cache and docker support too and the version 2.0 (whatever that is), is coming. The only drawback I see is the lack of MacOS VM. But we could probably leave that in Travis.

**Comment by ArnisLielturks** [comment]
Played around a bit with the semaphoreci. 
Linux GCC build took less than 11 minutes, on CircleCI it took a bit more than 14 minutes. Both built the project using 2 threads.
Semaphoreci: https://semaphoreci.com/arnislielturks/urho3d/branches/master/builds/6
CirleCI: https://circleci.com/gh/ArnisLielturks/Urho3D/80

One thing to note is that Semaphoreci documentation seemed a little bit harsh, hard to follow and it felt similar to Jenkins CI which I kinda stopped using since it required a lot of manual labour. For Travis and CircleCI there's one file configuration file saved in the repo, for semaphoreci I needed to write the build commands myself using their UI since I couldn't find how to do it differently but maybe I just didn't know where to look for it properly. I will do additional testing to get a bit more feedback.

**Comment by weitjong** [comment]
That's why I am waiting for version 2.0 https://semaphoreci.com/v2-is-coming

**Comment by weitjong** [comment]
I believe we can close this issue for now as I have put a quick fix to workaround the Travis issue. We can continue this discussion in #2348.
